### PR TITLE
Added Linaria for performance optimization

### LINKED ISSUE #5605: Optimize other field displays
## Scope & Context

Performance optimization on RecordTableCell

## Current behavior

We have already optimized RecordTableCell using RelationFieldDisplay as a baseline. 

## Expected behavior

We now want to apply all the relevant optimization on the other Field display components.

We'll work directly on a performance test for each field type, this method is called PTDD (performance test driven development)

```[tasklist]
### Tasks
- [x] ChipFieldDisplay
- [x] PhoneFieldDisplay
- [x] TextFieldDisplay
- [ ] UuidFieldDisplay
- [x] EmailFieldDisplay
- [x] DateTimeFieldDisplay
- [x] DateFieldDisplay
- [x] NumberFieldDisplay
- [ ] LinkFieldDisplay
- [ ] LinksFieldDisplay
- [x] CurrencyFieldDisplay
- [x] FullNameFieldDisplay
- [ ] SelectFieldDisplay
- [ ] MultiSelectFieldDisplay
- [ ] AddressFieldDisplay
- [ ] JsonFieldDisplay
```

### PR DESCRIPTION
**Pull request by lucasbordeau**
- Added Linaria to have compiled CSS on our optimized field displays
- Refactored mocks for performance stories on fields
- Refactored generateRecordChipData into a global context, computed only when we fetch object metadata items.
- Refactored ChipFieldDisplay 
- Refactored PhoneFieldDisplay

### CODE REVIEW

File: packages/twenty-front/src/testing/mock-data/peopleV2.ts [hunk 1]
```diff
@@ -0,0 +1,533 @@
+import { Company } from '@/companies/types/Company';
+import { Person } from '@/people/types/Person';
+
+export type MockedPersonV2 = Pick<
```
Comment (charlesBochet) [inline packages/twenty-front/src/testing/mock-data/peopleV2.ts:4]: why do we have a V2? Should we remove the other one?
We should not introduce V2 if there is an "easy way" to clean the v1. V2 is only relevant for big refactoring cases IMO

**Review by charlesBochet — COMMENTED** [review]
Linaria looks awesome, how are the performances?

File: packages/twenty-front/src/testing/mock-data/peopleV2.ts (hunk 1 above)
Comment (lucasbordeau) [reply packages/twenty-front/src/testing/mock-data/peopleV2.ts:4]: Refactored in next PR

**Comment by lucasbordeau** [comment]
@charlesBochet this PR should be merged first and then https://github.com/twentyhq/twenty/pull/5768

**Review by thomtrp — APPROVED** [review]

**Pull request merged**